Document GraphQL (#3704)
esl/MongooseIM: doc/graphql-api/Admin-GraphQL.md

> # MongooseIM's GraphQL API for the administration

Reviewer: maybe change "administration" to "administrator"?

> The new GraphQL admin API contains all the commands available through the REST API, and the vast majority of the CTL commands. Only commands that wouldn't have worked well with GraphQL style have been omitted.

Reviewer 2: I don't think that we've used the term "CTL" anywhere.

```suggestion
The new GraphQL admin API contains all the commands available through the REST API, and the vast majority of the CLI (`mongooseimctl`) commands. Only commands that wouldn't have worked well with GraphQL style have been omitted.
```

We can distinguish two levels of the administration. A global admin (has access to all commands), and the admin per domain (has access only to the own domain). Each of them is handled by a different endpoint. Please see the configuration [Listen](../../configuration/listen/#handler-types-graphql-api-mongoose_graphql_cowboy_handler) section for more details.

There is only one schema for both admin types. Admin per domain simply has no permissions to execute global commands or commands with not owned domain. The API documentation clearly says which commands are global.

> ## Domain per admin configuration

Reviewer 2: Either "domain admin" or "admin per domain"

> Out of the box, domains are created a with disabled admin account. Admin per domain can be enabled only by the global admin with the command

Reviewer 2: ```suggestion
Out of the box, domains are created with a disabled admin account. Admin per domain can be enabled only by the global admin with the command
```

<a href="../admin-graphql-doc.html#definition-DomainAdminMutation" target="_blank" rel="noopener noreferrer">mutation.domains.setDomainPassword</a>. Afterward, the domain admin can change the password with the same command.

The admin per domain can be disabled by the global admin with the command <a href="../admin-graphql-doc.html#definition-DomainAdminMutation" target="_blank" rel="noopener noreferrer">mutation.domains.removeDomainPassword</a>.

## Authentication

MongooseIM uses *Basic Authentication* as an authentication method for the GraphQL API.

*Basic authentication* is a simple authentication scheme built into the HTTP protocol.
Each HTTP request to the GraphQL API has to contain the Authorization header
with the word `Basic` followed by a space and a base64-encoded string.

### Global admin endpoint

The authentication for global admin is optional because this endpoint shouldn't be exposed outside. The credentials set in the handler section in the config enables the authentication. Please see the [GraphQL handler](../configuration/listen.md#handler-types-graphql-api-mongoose_graphql_cowboy_handler) section for more details.

> The base64-encoded string should have form

Reviewer 2: ```suggestion
The base64-encoded string should have the form
```
